core/internals.h

```diff
@@ -378,6 +430,34 @@ int senml_json_parse(const lwm2m_uri_t * uriP, const uint8_t * buffer, size_t bu
 int senml_json_serialize(const lwm2m_uri_t * uriP, int size, const lwm2m_data_t * tlvP, uint8_t ** bufferP);
 #endif
 
+#ifdef LWM2M_SUPPORT_SENML_CBOR
+// defined in cbor.c
+int cbor_get_type_and_value(const uint8_t *buffer, size_t bufferLen, cbor_type_t *type, uint64_t *value);
+int cbor_get_singular(const uint8_t *buffer, size_t bufferLen, lwm2m_data_t *dataP);
+int cbor_put_type_and_value(uint8_t *buffer, size_t bufferLen, cbor_type_t type, uint64_t val);
+int cbor_put_singular(uint8_t *buffer, size_t bufferLen, const lwm2m_data_t *dataP);
+#if !defined(LWM2M_VERSION_1_0) && !defined(LWM2M_VERSION_1_1)
+int cbor_parse(const lwm2m_uri_t *uriP, const uint8_t *buffer, size_t bufferLen, lwm2m_data_t **dataP);
+int cbor_serialize(const lwm2m_uri_t *uriP, int size, const lwm2m_data_t *dataP, uint8_t **bufferP);
+#endif
+
+// defined in senml_cbor.c
+int senml_cbor_parse(const lwm2m_uri_t *uriP, const uint8_t *buffer, size_t bufferLen, lwm2m_data_t **dataP);
+int senml_cbor_serialize(const lwm2m_uri_t *uriP, int size, const lwm2m_data_t *tlvP, uint8_t **bufferP);
+#endif
+
+// defined in senml_common.c
+#if defined(LWM2M_SUPPORT_JSON) || defined(LWM2M_SUPPORT_SENML_JSON) || defined(LWM2M_SUPPORT_SENML_CBOR)
+int senml_convert_records(const lwm2m_uri_t *uriP, senml_record_t *recordArray, int numRecords,
+                          senml_convertValue convertValue, lwm2m_data_t **dataP);
+lwm2m_data_t *senml_extendData(lwm2m_data_t *parentP, lwm2m_data_type_t type, uint16_t id);
+int senml_dataStrip(int size, lwm2m_data_t *dataP, lwm2m_data_t **resultP);
+lwm2m_data_t *senml_findDataItem(lwm2m_data_t *listP, size_t count, uint16_t id);
+uri_depth_t senml_decreaseLevel(uri_depth_t level);
+int senml_findAndCheckData(const lwm2m_uri_t *uriP, uri_depth_t baseLevel, size_t size, const lwm2m_data_t *tlvP,
+                           lwm2m_data_t **targetP, uri_depth_t *targetLevelP);
+#endif
```

Comment:
It is not really done at all in Wakaama, but having (at least) the return values documented would be reeeeeeealy nice. How is success encoded? >0? How an error? <0?